akka-http/src/main/scala/akka/http/javadsl/unmarshalling/sse/EventStreamUnmarshalling.scala

```diff
@@ -21,8 +22,31 @@ object EventStreamUnmarshalling {
   /**
    * Lets a `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of `ServerSentEvent`s.
    */
+  @deprecated("Use fromEventStreamWithTimeout", "10.0.10")
   val fromEventStream: Unmarshaller[HttpEntity, Source[ServerSentEvent, NotUsed]] =
     scaladsl.unmarshalling.sse.EventStreamUnmarshalling.fromEventStream
       .map(_.map(_.asInstanceOf[ServerSentEvent]).asJava)
       .asInstanceOf[Unmarshaller[HttpEntity, Source[ServerSentEvent, NotUsed]]]
+
+  /**
+   * Lets a `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of `ServerSentEvent`s.
+   *
+   * @param idleTimeout How long the connection should be allowed to be idle for before the connection is failed.
+   */
+  def fromEventStreamWithTimeout(idleTimeout: Timeout) = {
+    scaladsl.unmarshalling.sse.EventStreamUnmarshalling.fromEventStreamWithTimeout(idleTimeout)
+      .map(_.map(_.asInstanceOf[ServerSentEvent]).asJava)
+      .asInstanceOf[Unmarshaller[HttpEntity, Source[ServerSentEvent, NotUsed]]]
+  }
+
+  /**
+   * Lets a `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of `ServerSentEvent`s.
+   *
+   * Will shutdown the connection if the server doesn't send any events or heartbeats for 60 seconds.
+   */
+  def fromEventStreamWithTimeout() = {
```

Comment:
Hmmm, we could just add the timeout to the normal method, it'd change semantics a bit but in a helpful way I guess. And leave the methods be called fromEventStream, just adding the 1 param overload